src/components/MessageList/MessageList.jsx

```diff
@@ -1,43 +1,62 @@
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
-import './MessageList.css';
+
+import { loadMessages } from '../../store/Chat';
 import Message from '../Message/Message.jsx';
 
+import './MessageList.css';
+
 /**
  * @summary: responsible for handling the display of messages.
  */
 
-const MessageList = (props) => {
-  const { messages, activeId } = props;
-  const activeMsgs = messages.filter(msg => msg.chat === activeId);
+class MessageList extends Component {
+  static propTypes = {
+    messages: PropTypes.array.isRequired,
+    config: PropTypes.object.isRequired,
+    activeId: PropTypes.string,
+    dispatch: PropTypes.func.isRequired,
+  }
+
+  componentWillMount () {
+    const { activeId } = this.props;
+
+    if (this.props.config.apiServer && activeId) {
```

Comment:
I've noticed this pattern previous -- checking to see if the config's api server exists before making API calls -- maybe we can write a custom middle ware to do this ?